src/main.py

```diff
@@ -59,10 +59,10 @@ def __init__(self) -> None:
 if __name__ == "__main__":
     load_dotenv()
     aws = AWSData(
-        internal_role=os.environ["AWS_INTERNAL_ROLE"],
-        external_role=os.environ["AWS_EXTERNAL_ROLE"],
-        external_id=os.environ["AWS_EXTERNAL_ID"],
-        bucket=os.environ["AWS_BUCKET"],
+        internal_role=os.environ.get("AWS_INTERNAL_ROLE", "AWS_INTERNAL_ROLE"),
+        external_role=os.environ.get("AWS_EXTERNAL_ROLE", "AWS_EXTERNAL_ROLE"),
+        external_id=os.environ.get("AWS_EXTERNAL_ID", "AWS_EXTERNAL_ID"),
+        bucket=os.environ.get("AWS_BUCKET", "AWS_BUCKET"),
```

Comment:
What is the purpose of using fallback values here? They don't look like they would be useful in a program.
Should this maybe return an error if the keys don't exist?

Reply:
Problem is that they are read at the entry point of the program. And dry run mode doesn't actually require them. We wouldn't want to fail with a key error when we don't actually need them. 

I could move the reading of these environment variables to where they are needed, but then the program will run without knowing it's doesn't have what it needs. 

Maybe I will fall back to key error but put an if block here that doesn't try to read them in dry run mode. 

How does that sound? Other suggestions?

Reply:
>Maybe I will fall back to key error but put an if block here that doesn't try to read them in dry run mode.

If you already know at that point that you are running in `dry-run` mode this approach makes the most sense to me.

Reply:
Done!